File: src/baekjoon/p1193/solution.kt
//https://www.acmicpc.net/problem/1193
package baekjoon.p1193

fun main(args: Array<String>) {
    val num = readLine()!!.toInt();
    var edge = 0;
    var size = 1;
    while(edge < num){
        edge += size;
        size++;
    }
    size--;
    edge = edge - size;

    var upValue = when{
        size%2==0 -> num - edge;
        else -> size - (num - edge) + 1;
    }

    var downValue = when{
        size%2==0 -> size - (num - edge) + 1;
        else -> num - edge;

    }
    println(upValue.toString() + "/" + downValue.toString());
}

File: src/baekjoon/p1193/solution.py
#https://www.acmicpc.net/problem/1193
num = int(input())

size = 1
edge = 0

while edge < num:
    edge = edge + size
    size += 1

size -= 1
edge -= size

upValue = num - edge
downValue = num - edge
if size%2==0:
    downValue = size - downValue + 1

if size%2==1:
    upValue = size - upValue + 1

print(str(upValue) + '/' + str(downValue))
